contrib/gh-runner/gh-runner.sh

```diff
@@ -22,9 +22,26 @@ sudo usermod -aG docker ${USER}
 newgrp docker &
 # Install & Setup GH Actions Runner
 mkdir actions-runner && cd actions-runner
-curl -o actions-runner-linux-x64-2.296.2.tar.gz -L https://github.com/actions/runner/releases/download/v2.296.2/actions-runner-linux-x64-2.296.2.tar.gz
-echo "34a8f34956cdacd2156d4c658cce8dd54c5aef316a16bbbc95eb3ca4fd76429a  actions-runner-linux-x64-2.296.2.tar.gz" | shasum -a 256 -c
-tar xzf ./actions-runner-linux-x64-2.296.2.tar.gz
+if [ "$(uname -m)" = "aarch64" ]; then
+    echo "Detected arm64 architecture"
+    # Download the latest runner package
+    curl -o actions-runner-linux-arm64-2.298.2.tar.gz -L https://github.com/actions/runner/releases/download/v2.298.2/actions-runner-linux-arm64-2.298.2.tar.gz
+    # Optional: Validate the hash
+    echo "803e4aba36484ef4f126df184220946bc151ae1bbaf2b606b6e2f2280f5042e8  actions-runner-linux-arm64-2.298.2.tar.gz" | shasum -a 256 -c
+    # Extract the installer
+    tar xzf ./actions-runner-linux-arm64-2.298.2.tar.gz
+elif [ "$(uname -m)" = "x86_64" ]; then
+    echo "Detected amd64 architecture"
+    # Download the latest runner package
+    curl -o actions-runner-linux-x64-2.298.2.tar.gz -L https://github.com/actions/runner/releases/download/v2.298.2/actions-runner-linux-x64-2.298.2.tar.gz
+    # Optional: Validate the hash
+    echo "0bfd792196ce0ec6f1c65d2a9ad00215b2926ef2c416b8d97615265194477117  actions-runner-linux-x64-2.298.2.tar.gz" | shasum -a 256 -c 
+    # Extract the installer
+    tar xzf ./actions-runner-linux-x64-2.298.2.tar.gz
+else
+    echo "Unrecognized architecture"
```

Comment:
add `exit 1` here, I want the script to break when the architecture is unrecognized

Reply:
Added change